include/exadg/postprocessor/time_control.h

```diff
@@ -0,0 +1,59 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2021 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_COMPRESSIBLE_NAVIER_STOKES_POSTPROCESSOR_TIME_CONTROL_H_
+#define INCLUDE_COMPRESSIBLE_NAVIER_STOKES_POSTPROCESSOR_TIME_CONTROL_H_
+
+class TimeControl
+{
+public:
+  TimeControl();
+
+  void
+  reinit(bool const         calculate,
+         double const       start_time,
+         double const       calculate_every_time_interval,
+         unsigned int const calculate_every_time_steps);
+
+  /// If reinit() is not called needs_evaluation() will always evaluate to false
+  bool
+  needs_evaluation(double const time, unsigned int const time_step_number) const;
+
+  unsigned int
+  get_counter() const;
+
+  unsigned int
+  get_last_timestep_number_to_evaluate() const;
+
+private:
+  // small number which is much smaller than the time step size
+  double const EPSILON;
+  mutable bool reset_counter = true;
+
+  bool                 calculate;
```

Comment:
I wonder whether a slightly different name like `activate` would be better here, just because we are decoupling the time control from the "calculate" part in this PR and the name "calculate" might therefore no longer be clear.

Reply:
[89b7335](https://github.com/exadg/exadg/pull/262/commits/89b73350e0d87538534d8a900f21b72ff6b5abee)
should completely decouple from calculate